src/seedu/addressbook/ui/Formatter.java

```diff
@@ -0,0 +1,108 @@
+package seedu.addressbook.ui;
+
+import static seedu.addressbook.common.Messages.MESSAGE_GOODBYE;
+import static seedu.addressbook.common.Messages.MESSAGE_INIT_FAILED;
+import static seedu.addressbook.common.Messages.MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE;
+import static seedu.addressbook.common.Messages.MESSAGE_USING_STORAGE_FILE;
+import static seedu.addressbook.common.Messages.MESSAGE_WELCOME;
+
+import java.io.PrintStream;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Scanner;
+
+import seedu.addressbook.commands.CommandResult;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+public class Formatter {
+	public Scanner in;
+	public PrintStream out;
+
+	public Formatter() {
+	}
+
+	public void showWelcomeMessage(TextUi textUi, String version, String storageFilePath) {
+	    String storageFileInfo = String.format(MESSAGE_USING_STORAGE_FILE, storageFilePath);
+	    textUi.formatter.showToUser(
+	            textUi, TextUi.DIVIDER,
+	            TextUi.DIVIDER,
+	            MESSAGE_WELCOME,
+	            version,
+	            MESSAGE_PROGRAM_LAUNCH_ARGS_USAGE,
+	            storageFileInfo,
+	            TextUi.DIVIDER);
+	}
+
+	public void showGoodbyeMessage(TextUi textUi) {
+	    textUi.formatter.showToUser(textUi, MESSAGE_GOODBYE, TextUi.DIVIDER, TextUi.DIVIDER);
+	}
+
+	public void showInitFailedMessage(TextUi textUi) {
+	    textUi.formatter.showToUser(textUi, MESSAGE_INIT_FAILED, TextUi.DIVIDER, TextUi.DIVIDER);
+	}
+
+	/** Shows message(s) to the user 
+	 * @param textUi TODO
+	 * @param message TODO*/
+	public void showToUser(TextUi textUi, String... message) {
```

Comment:
your formatter is also doing printing work; essentially violating SRP, albeit in a different way! please think how you can eliminate this problem; recollect the discussion around this during the tutorial :)